lte/gateway/c/session_manager/PipelinedClient.h

```diff
@@ -155,8 +155,10 @@ class PipelinedClient {
       const SessionState::SessionInfo info,
       std::function<void(Status status, UPFSessionContextState)> callback) = 0;
 
-  virtual uint32_t get_next_teid()    = 0;
-  virtual uint32_t get_current_teid() = 0;
+  virtual uint32_t get_next_teid()               = 0;
+  virtual uint32_t get_current_teid()            = 0;
+  virtual void set_upf_n3_addr(std::string addr) = 0;
+  virtual std::string get_upf_n3_addr()          = 0;
```

Comment:
is pipelined providing new TEIDs? I thought teids would come from AMF and pipelined would get those from sessiond

Reply:
Well for Uplink traffic AMF provide TEID's to UPF, however for downlink traffic SMF provides the teid's to gnodeb, and the same used to encapsulated by  UPF. As it is linked to specific UPF client, we added as part of pipelined code.  There is no better place to add right ?